### PR TITLE
Undefine some symbols and globalOptions when processing nimscript

### PR DESCRIPTION
**Pull request by nc-x**
Fixes https://github.com/nim-lang/Nim/issues/4446
Fixes https://github.com/nim-lang/Nim/issues/8991
Fixes https://github.com/nim-lang/Nim/issues/9420
Fixes https://github.com/nim-lang/Nim/issues/9589
Fixes https://github.com/nim-lang/Nim/issues/9153
Fixes https://github.com/nim-lang/Nim/issues/9120

Note that there is no need to re-enable these symbols because a few lines later we already re-process the command line argumesnts -
https://github.com/nim-lang/Nim/blob/b104f77444beb3a19732ad5d7728ce2109325be9/compiler/cmdlinehelper.nim#L91

### CODE REVIEW

**Comment by nc-x** [comment]
This should be ready now.

File: compiler/cmdlinehelper.nim [hunk 1]
```diff
@@ -48,6 +48,15 @@ proc loadConfigsAndRunMainCommand*(self: NimProg, cache: IdentCache; conf: Confi
   if self.suggestMode:
     conf.command = "nimsuggest"
 
+  # These defines/options should not be enabled while processing nimscript
+  # bug #4446, #9420, #8991, #9589, #9153
+  var symsToUndef = ["profiler", "memProfiler", "nodejs"]
```
Comment (data-man) [inline compiler/cmdlinehelper.nim]: var -> const

File: compiler/cmdlinehelper.nim [hunk 2]
```diff
@@ -48,6 +48,15 @@ proc loadConfigsAndRunMainCommand*(self: NimProg, cache: IdentCache; conf: Confi
...
+  for sym in symsToUndef:
```
Comment (data-man) [inline compiler/cmdlinehelper.nim]: Unroll loop?
```nim
undefSymbol(conf.symbols, "profiler")
undefSymbol(conf.symbols, "memProfiler")
undefSymbol(conf.symbols, "nodejs")```

**Comment by nc-x** [comment]
Done.

**Pull request merged**